modules/dnn/src/layers/eltwise_layer.cpp

```diff
@@ -64,15 +64,6 @@ namespace dnn
 class EltwiseLayerImpl CV_FINAL : public EltwiseLayer
 {
 public:
-    enum EltwiseOp
-    {
-        PROD = 0,
-        SUM = 1,
-        MAX = 2,
-        DIV = 3
-    } op;
-    std::vector<float> coeffs;
-
     enum OutputChannelsMode
```

Comment:
I have moved `EltwiseOp`, `op` and `coeffs` to `EltwiseLayer` (which is a part of the public headers). This will alter the layout of `EltwiseLayer` and hence is not ABI compatible. Is this ok?

Reply:
Public API of DNN layers is `LayerParams`.

Reply:
> all_layers.hpp

Remove changes from there. This header is part of public API.

If you need to access to EltwiseLayerImpl, then create internal `modules/dnn/src/layers/eltwise_layer.hpp`

Reply:
I need access to just `op` and `coeffs`. The CUDA backend can fuse eltwise if:
- the operation is `SUM`
- has no coefficients
- no variable channels

The variable channels is easy to check as the input shapes are available in `fuseLayers()`. I moved `op` and `coeffs` to `EltwiseLayer` (which  is in `all_layers.hpp`) so that they are accessible in `fuseLayers`.

I tried to see what the OCL code was doing to check the conditions but I couldn't find any such checks. Maybe I can use the same technique as OCL code to get those properties.